### PR TITLE
Wasserstein loss function

### PR DESCRIPTION
**Pull request by rnett**
## What changes were proposed in this pull request?

Added the Wasserstein loss function.

Forward pass: `mean(labels * output, dims=1)`
Backwards pass: `labels/labels.shape(1)`

## How was this patch tested?

None.  Should I add tests?  I didn't see tests for any other losses except for crossentropy.

Note that the gradient doesn't match SameDiff's b/c of #7405.

### CODE REVIEW

**Comment by rnett** [comment]
Note that the improved Wasserstein loss is better, but requires the 2nd degree gradient, which I don't think we can calculate?  Specifically, it adds a `gradient - 1` term to the loss.

**Comment by rnett** [comment]
Tests are in and the gradients pass.

The tests fail the serialization test because of an extra space at the end of the expected serialized string here:
```
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:118)
	at org.junit.Assert.assertEquals(Assert.java:144)
	at org.deeplearning4j.TestUtils.testModelSerialization(TestUtils.java:63)
	at org.deeplearning4j.gradientcheck.LossFunctionGradientCheck.lossFunctionGradientCheckLossLayer(LossFunctionGradientCheck.java:373)
```

**Comment by AlexDBlack** [comment]
> The tests fail the serialization test because of an extra space at the end of the expected serialized string 

I haven't seen that before in any of the tests :/
I'll check out and build locally

**Comment by AlexDBlack** [comment]
OK, so the problem here is the equals method in DifferentialFunction.
When the equality check fails, it does toString which returns network JSON - that extra space isn't relevant here.
Let's just switch to `@EqualsAndHashCode(callSuper = false)` - superclass fields aren't relevant/used, so we can ignore them. Test passes with that.
